app/models/pxe_server.rb

```diff
@@ -198,6 +198,14 @@ def delete_provisioning_files(pxe_image, mac_address, windows_image = nil, custo
     _log.info("#{log_message}...Complete")
   end
 
+  def ensure_menu_list(menu_list)
+    current_menus = pxe_menus.map(&:file_name)
+    to_destroy = current_menus.reject { |menu| menu_list.include?(menu) }
+    to_create = menu_list.reject { |menu| current_menus.include?(menu) }
+    pxe_menus.where(:file_name => to_destroy).destroy_all
+    to_create.each { |menu| pxe_menus << PxeMenu.create(:file_name => menu) }
```

Comment:
- Should this be `create!`?
- I think it might be better to wrap the destroy line and this line into a transaction, so that there aren't race conditions.

Reply:
Thank you, done. We're invoking the whole thing in a transaction anyway (https://github.com/ManageIQ/manageiq-api/pull/648/files#diff-36ba675b6b089aade991c039ccf18852R50) but you're right, this is a standalone function and we need to ensure it'll work either way